### PR TITLE
cli(ticdc): remove dead code

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4954

### What is changed and how it works?

- Remove useless check func
- Remove useless flag

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integration_tests/http_api/run.sh [hunk 1]
```diff
@@ -25,15 +25,15 @@ function run() {
...
 	owner_pid=$(ps -C $CDC_BINARY -o pid= | awk '{print $1}')
-	owner_id=$($CDC_BINARY cli capture list --disable-version-check 2>&1 | awk -F '"' '/id/{print $4}')
+	owner_id=$($CDC_BINARY cli capture list 2>&1 | awk -F '"' '/id/{print $4}')
 	echo "owner pid:" $owner_pid
 	echo "owner id" $owner_id
 
 	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --addr "127.0.0.1:8301"
-	ensure $MAX_RETRIES "$CDC_BINARY cli capture list --disable-version-check 2>&1 | grep -v \"$owner_id\" | grep id"
-	capture_id=$($CDC_BINARY cli capture list --disable-version-check 2>&1 | awk -F '"' '/id/{print $4}' | grep -v "$owner_id")
+	ensure $MAX_RETRIES "$CDC_BINARY cli capture list  2>&1 | grep -v \"$owner_id\" | grep id"
```
Comment (zhaoxinyu) [inline tests/integration_tests/http_api/run.sh:35]:
```suggestion
	ensure $MAX_RETRIES "$CDC_BINARY cli capture list 2>&1 | grep -v \"$owner_id\" | grep id"
```

File: tests/integration_tests/http_api/run.sh [hunk 2]
```diff
@@ -25,15 +25,15 @@ function run() {
 
 	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY
 	# wait for cdc run
-	ensure $MAX_RETRIES "$CDC_BINARY cli capture list --disable-version-check 2>&1 | grep '\"is-owner\": true'"
+	ensure $MAX_RETRIES "$CDC_BINARY cli capture list  2>&1 | grep '\"is-owner\": true'"
```
Comment (zhaoxinyu) [inline tests/integration_tests/http_api/run.sh:28]:
```suggestion
	ensure $MAX_RETRIES "$CDC_BINARY cli capture list 2>&1 | grep '\"is-owner\": true'"
```

File: tests/integration_tests/availability/processor.sh [hunk 3]
```diff
@@ -21,14 +21,14 @@ function test_stop_processor() {
 	# start a capture server
 	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --logsuffix test_stop_processor
 	# ensure the server become the owner
-	ensure $MAX_RETRIES "$CDC_BINARY cli capture list --disable-version-check 2>&1 | grep '\"is-owner\": true'"
+	ensure $MAX_RETRIES "$CDC_BINARY cli capture list  2>&1 | grep '\"is-owner\": true'"
```
Comment (zhaoxinyu) [inline tests/integration_tests/availability/processor.sh:24]:
```suggestion
	ensure $MAX_RETRIES "$CDC_BINARY cli capture list 2>&1 | grep '\"is-owner\": true'"
```

File: tests/integration_tests/availability/owner.sh [hunk 4]
```diff
@@ -211,17 +211,17 @@ function test_gap_between_watch_capture() {
...
 	owner_pid=$(ps -C $CDC_BINARY -o pid= | awk '{print $1}')
-	owner_id=$($CDC_BINARY cli capture list --disable-version-check 2>&1 | awk -F '"' '/id/{print $4}')
+	owner_id=$($CDC_BINARY cli capture list 2>&1 | awk -F '"' '/id/{print $4}')
 	echo "owner pid:" $owner_pid
 	echo "owner id" $owner_id
 
 	# run another server
 	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --addr "127.0.0.1:8301" --logsuffix test_gap_between_watch_capture.server2
-	ensure $MAX_RETRIES "$CDC_BINARY cli capture list --disable-version-check 2>&1 | grep -v \"$owner_id\" | grep id"
+	ensure $MAX_RETRIES "$CDC_BINARY cli capture list  2>&1 | grep -v \"$owner_id\" | grep id"
```
Comment (zhaoxinyu) [inline tests/integration_tests/availability/owner.sh:222]:
```suggestion
	ensure $MAX_RETRIES "$CDC_BINARY cli capture list 2>&1 | grep -v \"$owner_id\" | grep id"
```

**Review by zhaoxinyu — APPROVED** [review]

File: tests/integration_tests/availability/owner.sh (hunk 4 above)
Comment (zhaoxinyu) [reply tests/integration_tests/availability/owner.sh:222]: Rest LGTML

**Review by sdojjy — APPROVED** [review]

**Pull request merged**